src/workspaceConfig.ts

```diff
@@ -45,16 +44,27 @@ export function updateIdfComponentsTree(projectDescriptionPath: string) {
     idfDataProvider.refresh(projectDescriptionPath);
 }
 
-export function updateProjectName(projectDescriptionPath: string) {
-    if (!utils.fileExists(projectDescriptionPath)) {
-        return;
-    }
-    fs.readFile(projectDescriptionPath, (err, data) => {
-        if (err) {
-            Logger.errorNotify(err.message, err);
-            return;
+export function getProjectName(workspace: string): Promise<string> {
+    const cmakeListFile = path.join(workspace, "CMakeLists.txt");
+    return new Promise((resolve, reject) => {
+        try {
+            if (!utils.fileExists(cmakeListFile)) {
+                reject(`${cmakeListFile} doesn't exist.`);
+            }
+            fs.readFile(cmakeListFile, (err, data) => {
+                if (err) {
+                    Logger.errorNotify(err.message, err);
+                    reject(err);
+                }
+                const content = data.toString();
+                const projectMatches = content.match(/(?:project\()(.*?)(?:\))/);
+                if (projectMatches && projectMatches.length > 0) {
+                    resolve(projectMatches[1]);
+                }
```

Comment:
else if missing !

if the condition is not satisfies the promise will not be rejected, neither be resolved, leading to an indeterministic state